### DISCUSSION TITLE
[Docs Improvement]: Add Tailwind CLI documentation

### DISCUSSION THREAD

**Discussion by d4h0** [tailwindlabs/tailwindcss#2240] [Ideas]
At the moment the installation page says:

> Use the npx tailwindcss help build command to learn more about the various CLI options.

Which isn't great if I don't have tailwindcss installed already.

I think it would be more user friendly if the documentation was available online.

Because now a potential user has to install tailwindcss to see what the CLI can do (and can't).

I guess some potential users who are interested in the CLI are not familiar with the Node.js ecosystem, so for them, the installation could already be a big obstacle.

Therefore it would be nice if these users could check out the functionality etc. of the CLI without installing it.

**Comment by simonswiss** [comment]
Thanks for this suggestion, @d4h0, you're making some great points 👍

We'll be working on a series of installation guides in an upcoming work cycle, and might revisit some of the current documentation on the installation page.

Making a note internally to consider improving the "onboarding journey" for CLI users.

Will move this to our community discussions!

**Comment by justingolden21** [comment]
It's late 2022 and we still don't have any docs (that I can find anyway) on the CLI...

https://tailwindcss.com/docs/tailwind-cli

![image](https://user-images.githubusercontent.com/30274440/184261029-5dab2e08-187a-451a-9b3e-e6f83d172fa8.png)

This is the ENTIRE PAGE... That's crazy.

This page https://tailwindcss.com/docs/installation has some basic setup, but doesn't document the flags or anything at all.

**Reply by jasonhulbert** [reply]
```tailwindcss --help```

**Comment by williamisnotdefined** [comment]
tailwindcss v4: `npx @tailwindcss/cli --help`

**Reply by topherfangio** [reply]
For reference, this is what it currently outputs; which isn't terribly helpful.

```
root@784a8e25efac:/home/loco_demo# npx @tailwindcss/cli --help
≈ tailwindcss v4.0.12

Usage:
  tailwindcss [--input input.css] [--output output.css] [--watch] [options…]

Options:
  -i, --input ··········· Input file
  -o, --output ·········· Output file [default: `-`]
  -w, --watch ··········· Watch for changes and rebuild as needed
  -m, --minify ·········· Optimize and minify the output
      --optimize ········ Optimize the output without minifying
      --cwd ············· The current working directory [default: `.`]
  -h, --help ············ Display usage information
  ```

Is there debug mode so we can see more info when things aren't working? Tailwind 4 is only applying some of my classes (like `text-4xl` but not `text-5xl` for some reason).

It would be GREAT if we could see what source files / paths it was looking at to make sure it wasn't missing something.

**Comment by dslatkin** [comment]
There is a guide in [this thread](https://github.com/tailwindlabs/tailwindcss/discussions/15855#discussion-7869567) that has been helpful for me. That and inspecting the CLI's main file at `packages/@tailwindcss-cli/src/commands/build/index.ts`.
